woof-code/rootfs-packages/pup-advert-blocker/usr/sbin/pup-advert-blocker

```diff
@@ -0,0 +1,163 @@
+#!/bin/sh
+#v0.3 created by sc0ttman, August 2010: http://murga-linux.com/puppy/viewtopic.php?t=59290
+#GPL license /usr/share/doc/legal/gpl-2.0.txt
+#100830 BK added GPL license, amended Exit msg, bug fixes.
+#120202 rodin.s: adding gettext
+#140212 zigbert: gui (gtkdialog) improvements.
+
+# advert blocker
+# downloads a list of known advert servers
+# then appends them to /etc/hosts so that
+# many online adverts are blocked from sight
+
+# make a hosts file if none found, or add a marker
+
+export TEXTDOMAIN=pup-advert-blocker
+export TEXTDOMAINDIR=/usr/share/locale
+export OUTPUT_CHARSET=UTF-8
+eval_gettext () {
+  local myMESSAGE=$(gettext "$1")
+  eval echo \"$myMESSAGE\"
+}
+export LANGORG=$LANG
+
+if [ ! -f /etc/hosts ];then 
+	echo "#host file
+127.0.0.1 localhost puppypc
+" > /etc/hosts
+fi
+
+# set vars
+export appver='0.3'
+export title='Pup Advert Blocker'
+# the markers used to find the changes in /etc/hosts, which are made by this app
+export markerstart='# pup-advert-blocker IPs below'
+export markerend='# pup-advert-blocker IPs above'
+
+# create functions
+set -a
+
+# cleanup all leftover files
+cleanup () {
+	# remove all temp files
+	rm -f /tmp/adlist-all /tmp/hosts-temp
+}
+
+# download the ads list
+download_adlist () {
+	wget -t 2 -O- https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts | grep '^0\.0\.0\.0' | grep -v '^0\.0\.0\.0 [0-9]*\.[0-9]*\.[0-9]*\.[0-9]*$' > /tmp/adlist-all
```

Comment:
@sc0ttj I've changed it so it uses a ready-made, combined hosts list (malware + ads)